### ISSUE TITLE
Grouped box plot with multiple variables

### ISSUE THREAD

**Issue by guybrettrobertson** [JohnCoene/echarts4r#409]
**Is there a way to create a grouped box plot with multiple grouping variables using {echarts4r}?**

I would like to create a grouped box plot with multiple variables comparable to this {ggplot2} plot: https://www.r-graph-gallery.com/265-grouped-boxplot-with-ggplot2.html
![unnamed-chunk-1-1](https://user-images.githubusercontent.com/65654871/155708510-6c675448-adb5-4968-9e98-e1f11b368d15.png)

I have tried the following:
```
variety = rep(LETTERS[1:7], each=40)
treatment = rep(c("high","low"),each=20)
note = seq(1:280)+sample(1:150, 280, replace=T)

data.frame(variety, treatment ,  note) |>
  echarts4r::group_by(variety, treatment) |>
  echarts4r::e_charts() |>
  echarts4r::e_boxplot(note) |>
  echarts4r::e_legend()
```

But it gives me this:
![Rplot](https://user-images.githubusercontent.com/65654871/155708742-b29810f0-b6a4-451a-9795-d5cae781ec0a.png)

I have also tried using `e_add_nested` but it doesn't seem to work for boxplots.

**Comment by guybrettrobertson** [comment]
I just noticed that has already come up in issue 253 but no clean solution was available at the time. Have any developments been made on this?

**Comment by sandra-ab** [comment]
I would love this feature too. I've been using bar charts as a workaround (where the multiple grouping works as expected), but often boxplots are just so much more appropriate. :)

**Comment by JohnCoene** [comment]
#253

**Comment by munoztd0** [comment]
A "semi-complicated" workaround

https://github.com/JohnCoene/echarts4r/issues/253#issuecomment-1204012312

**Comment by sharp-lilac** [comment]
Hi, any news on this? Boxplots are so important for data visualizations and not being able to group by multiple variables is really surprising. Any workarounds?

**Comment by munoztd0** [comment]
@sharp-lilac  I agree with you is just that boxplots are not "normal" plots they are constituted of various diverse source of information (aggregated statistics as well a raw data points, i.e. outliers.)

I started devloping a new branch that will implement a new approach for boxplots but it might get ugly at first, the commit 06c2e5d68cb507170fa6ff89157a083ae41373d8 shows the first steps.

On the branch grouped_boxplot:

``` r
library(echarts4r)

data <- data.frame(
  category = rep(LETTERS[1:5], each = 100),
  value = rnorm(500, mean = rep(1:5, each = 100), sd = 1:5),
  gender = sample(c("M", "F"), 500, replace = TRUE)
)

plot <- data |>
  group_by(gender) |>
  e_charts(category) |>
  e_boxplot_group_(serie = "value") |>
  e_x_axis(name = "Category") |>
  e_y_axis(name = "Value") |>
  e_tooltip(trigger = "axis") |>
  e_legend(show = TRUE)

plot
```

![image](https://github.com/user-attachments/assets/cfb6ee96-7367-4da5-b704-e286b5e4cb9f)


<sup>Created on 2024-11-25 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by sharp-lilac** [comment]
@munoztd0 Very cool thank you! I can't get the function to generate outliers, though. 

I get this error:

Error in .add_outliers(e, serie, i) : 
  could not find function ".add_outliers"

And if I remove the problematic line in the function, I can generate grouped boxplots great, but without outliers. Any idea why? I just copied the new function into my script and ran your example code.

**Comment by munoztd0** [comment]
Hey, glad it kinda works.
Yeah I know, I haven't finished implementing the outliers. I have to refactor it somehow, work in progress.
